test/test_torch.py

```diff
@@ -9581,6 +9581,44 @@ def test_lerp(self, device):
                 expected = start + weight * (end - start)
                 self.assertEqual(expected, actual)
 
+    def _test_logaddexp(self, device, dtype, base2=False):
+        if base2:
+            gt_func = np.logaddexp2
+            our_func = torch.logaddexp2
+        else:
+            gt_func = np.logaddexp
+            our_func = torch.logaddexp
+
+        a = torch.randn(51, 2, dtype=dtype, device=device)
+        b = torch.randn(51, 2, dtype=dtype, device=device)
+        gt = torch.tensor(gt_func(a.cpu().numpy(), b.cpu().numpy()), device=device)
+        ours = our_func(a, b)
+        self.assertEqual(ours.shape, gt.shape)
+        self.assertTrue(torch.allclose(ours, gt))
+
+        # numerical stability
+        a *= 10000
+        b *= 10000
+        a[0] = inf
```

Comment:
It's really great how thorough you're being here. The sample code should simplify this further.